### ISSUE TITLE
Race condition on initial fresh database startup and no mute role causes exception

### ISSUE THREAD

**Issue by Inzanit** [discord-csharp/MODiX#527]
A race condition as part of the ModerationService both creating (and attempting to use) the mute role, when it doesn't exist, causes several exceptions on startup. MODiX **does** start up, however and the role does get created.

See:

https://github.com/discord-csharp/MODiX/blob/4e704f4b039ac07ac6413b4a4839aba163d8bb8b/Modix.Services/Moderation/ModerationService.cs#L753-L756

Associated exception:

```[22:44:06 ERR] Failed executing DbCommand (9ms) [Parameters=[@p7='?' (DbType = Int64), @p8='?' (DbType = Int64), @p9='?' (DbType = Int64), @p10='?' (DbType = Int64), @p11='?'], CommandType='Text', CommandTimeout='30']
INSERT INTO "DesignatedRoleMappings" ("CreateActionId", "DeleteActionId", "GuildId", "RoleId", "Type")
VALUES (@p7, @p8, @p9, @p10, @p11)
RETURNING "Id";
Npgsql.PostgresException (0x80004005): 23503: insert or update on table "DesignatedRoleMappings" violates foreign key constraint "FK_DesignatedRoleMappings_GuildRoles_RoleId"
   at Npgsql.NpgsqlConnector.<>c__DisplayClass161_0.<<ReadMessage>g__ReadMessageLong|0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Npgsql.NpgsqlConnector.<>c__DisplayClass161_0.<<ReadMessage>g__ReadMessageLong|0>d.MoveNext() in C:\projects\npgsql\src\Npgsql\NpgsqlConnector.cs:line 1032
--- End of stack trace from previous location where exception was thrown ---
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming) in C:\projects\npgsql\src\Npgsql\NpgsqlDataReader.cs:line 489
   at Npgsql.NpgsqlCommand.ExecuteDbDataReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken) in C:\projects\npgsql\src\Npgsql\NpgsqlCommand.cs:line 1215
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteAsync(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues, CancellationToken cancellationToken)````

**Comment by Foxtrek64** [comment]
Can confirm the existence of this bug. Just experienced it a few minutes ago when joining MODiX to a new server.

Changing someone's nickname seems to "force" MODiX to figure itself out, though how this happens I'm not entirely sure. I've not dug around into the db side of things.

Either way, there's a temporary work-around for those experiencing this issue.

**Comment by Sutarodo** [comment]
@Foxtrek64 Try again. Seems it was fixed and this can be closed.

**Comment by Inzanit** [comment]
This is fixed as of #528
